server/src/main/java/io/crate/metadata/SystemTable.java

```diff
@@ -397,7 +397,14 @@ public P endObject() {
         }
 
         private <U> void addColumnToParent(Column<T, U> column) {
-            parent.add(new Column<>(column.column, column.type, column.getProperty));
+            if (column instanceof DynamicColumn dynamicColumn) {
+                parent.add(new DynamicColumn<>(
+                    dynamicColumn.column,
+                    dynamicColumn.leafType,
+                    t -> objectIsNull.test(t) ? Map.of() : (Map<String, Object>) dynamicColumn.getProperty.apply(t)));
+            } else {
+                parent.add(new Column<>(column.column, column.type, column.getProperty));
+            }
```

Comment:
Why do we create a new column instance in general? Couldn't we just add existing instance?

Reply:
Thank you, I noticed by did not look into it. I will do that now.

Reply:
Updated, I think they don't need to be recreated.